src/WorldIDIdentityManagerImplV3.sol

```diff
@@ -0,0 +1,302 @@
+    ///         registrations.
+    /// @dev Only the owner of the contract can call this function.
+    ///
+    /// @param newTable The new verifier lookup table to be used for verifying identity
+    ///        registrations.
+    /// @custom:reverts InvalidVerifierLUT if `newTable` is set to the zero address
+    function setRegisterIdentities4844VerifierLookupTable(VerifierLookupTable4844 newTable)
+        public
+        virtual
+        onlyProxy
+        onlyInitialized
+        onlyOwner
+    {
+        if (address(newTable) == address(0)) {
+            revert InvalidVerifierLUT();
+        }
+
+        VerifierLookupTable4844 oldTable = batchInsertion4844Verifiers;
+        batchInsertion4844Verifiers = newTable;
+        emit DependencyUpdated(
+            Dependency.InsertionVerifierLookupTable, address(oldTable), address(newTable)
+        );
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                               PUBLIC TYPES                              ///
+    ///////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Parameters for registerIdentities4844 function.
+    /// @dev This struct holds the parameters for registering identities and verifying the insertion proof
+    ///      using KZG proofs as described in EIP-4844.
+    struct RegisterIdentities4844Params {
+        /// @notice The proof that given the conditions, insertion into the tree results in `postRoot`.
+        ///         Elements 0 and 1 are the `x` and `y` coordinates for `ar` respectively. Elements 2 and 3 are the
+        ///         `x` coordinate for `bs`, and elements 4 and 5 are the `y` coordinate for `bs`. Elements 6 and 7
+        ///         are the `x` and `y` coordinates for `krs`.
+        uint256[8] insertionProof;
+        /// @notice The Pedersen commitments from the proof.
+        uint256[2] commitments;
+        /// @notice The proof of knowledge for the Pedersen commitments.
+        uint256[2] commitmentPok;
+        /// @notice KZG commitment for the polynomial extrapolated from the identities.
+        uint128[3] kzgCommitment;
+        /// @notice KZG proof associated with the commitment.
+        uint128[3] kzgProof;
+        /// @notice Expected evaluation of the polynomial at a certain point equal to KZG challenge.
+        uint256 expectedEvaluation;
+        /// @notice The value for the root of the tree before the `identityCommitments` have been inserted.
+        ///         Must be an element of the field `Kr`. (already in reduced form)
+        uint256 preRoot;
+        /// @notice The root obtained after inserting all of `identityCommitments` into the tree described
+        ///         by `preRoot`. Must be an element of the field `Kr`. (already in reduced form)
+        uint256 postRoot;
+        /// @notice Root of the merkle tree after adding identities to it.
```

Comment:
This is misleading. Better wording: `Hash of all inserted identities, constructed by taking a root of the Merkle Tree of minimal depth containing all identities`.